text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,1655 @@
+         * cdk init does not have to support template configuration
+           * If template authors want to have variations on templates, they can do the branching generation-side, which allows them to use more advanced tools
+       * Cons:
+         * No default placeholder or template configuration style for template authors
+
+
+
+### What are the drawbacks of this solution?
+
+The template validation burden is on template author, so we can't ensure template quality for CDK users unless the custom template is part of vetted CLI template registry.
+
+
+
+### What is the high-level project plan?
+
+* Phase 1: Core functionality
+
+  * Support for custom templates through local files, Git repositories, and NPM packages
+  * Support for custom templates in all CDK languages (TypeScript, C#, F#, Go, Java, JavaScript, and Python)
+  * Support for public and private Git repositories and NPM packages
+  * Support for users to specify what branch/commit of repo or package version they want to pull templates from and if not specified then will use main branch/most recent commit or latest package version by default
+  * Basic custom template validation which checks that template follows schema defined in "Authoring Custom Templates" section of RFC
+
+* Phase 2: Testing and Static Discovery for Custom Templates
+
+  * Develop unit and integration tests
+  * Extend `cdk init --list` to display public Git repositories and NPM packages for custom templates, created by internal AWS teams and vetted by CDK team.
+  * Work with internal AWS teams to create and add more templates to the CDK CLI registry
+
+* Phase 3: Support for additional NPM package formats
+
+  * Extend `--from-npm` to support tarball formats:
+    * Local gzipped tarballs: `cdk init --from-npm ./my-template.tgz --language=typescript`
+    * Remote tarball URLs: `cdk init --from-npm https://example.com/my-template.tgz --language=typescript`
+
+* Phase 4: Documentation, testing, and marketing
+  * Publish official documentation ("Authoring Custom CDK Templates" section) to docs.aws.amazon.com/cdk/ for template authors to reference
+  * Publish official documentation ("Updated README for Users of CDK CLI" section) to github.com/aws/aws-cdk-cli/tree/main/packages/aws-cdk for CDK users to reference
+  * Author blog post and other marketing materials
+    * Include benefits and example use cases of passing in custom templates through Local/Git/NPM
+    * Include benefits and example use cases of selecting branch/commit/package version of a custom template
+
+### Are there any future enhancements of this feature?
+
+* Publish sample templates on GitHub or NPM for template authors to reference when creating
+* Dynamic template discovery mechanism to browse all available public template beyond static discovery in CLI
+* Integrate CDK CLI telemetry to track which sources for custom templates are used and how often
+
+
+
+## Appendix
+
+### Example CDK App Templates for Supported Languages
```

Comment:
I'm still not super convinced about the utility of this section. Could it be part of your sample repo?